src/client.cpp

```diff
@@ -395,27 +395,18 @@ void Client::step(float dtime)
 
 			Player *myplayer = m_env.getLocalPlayer();
 			assert(myplayer != NULL);
-			// Send TOSERVER_INIT
-			// [0] u16 TOSERVER_INIT
-			// [2] u8 SER_FMT_VER_HIGHEST_READ
-			// [3] u8[20] player_name
-			// [23] u8[28] password (new in some version)
-			// [51] u16 minimum supported network protocol version (added sometime)
-			// [53] u16 maximum supported network protocol version (added later than the previous one)
 
-			char pName[PLAYERNAME_SIZE];
-			char pPassword[PASSWORD_SIZE];
+			NetworkPacket* pkt = new NetworkPacket(TOSERVER_INIT, 0);
 
-			snprintf(pName, PLAYERNAME_SIZE, "%s", myplayer->getName());
-			snprintf(pPassword, PASSWORD_SIZE, "%s", m_password.c_str());
+			// Set the supported compression modes here (8 modes possible)
+			u8 compression_mode = 0;
+			compression_mode |= (1 << NETPROTO_COMPRESSION_NONE);
```

Comment:
I don't think that no compression needs a bit; i.e. 0 would be no compression. Therefore until another compression method is supported, the default value of compression_mode should simply be `(1 << NETPROTO_COMPRESSION_ZLIB)`